Lesson23/Feeling_lucky.py

```diff
@@ -0,0 +1,240 @@
+<li> <a href="http://udacity.com/cs101x/urank/arsenic.html">World's Best Hummus</a>
+<li> <a href="http://udacity.com/cs101x/urank/kathleen.html">Kathleen's Hummus Recipe</a>
+</ul>
+
+For more expert opinions, check out the
+<a href="http://udacity.com/cs101x/urank/nickel.html">Nickel Chef</a>
+and <a href="http://udacity.com/cs101x/urank/zinc.html">Zinc Chef</a>.
+</body>
+</html>
+
+
+
+
+
+
+""",
+   'http://udacity.com/cs101x/urank/zinc.html': """<html>
+<body>
+<h1>The Zinc Chef</h1>
+<p>
+I learned everything I know from
+<a href="http://udacity.com/cs101x/urank/nickel.html">the Nickel Chef</a>.
+</p>
+<p>
+For great hummus, try
+<a href="http://udacity.com/cs101x/urank/arsenic.html">this recipe</a>.
+
+</body>
+</html>
+
+
+
+
+
+
+""",
+   'http://udacity.com/cs101x/urank/nickel.html': """<html>
+<body>
+<h1>The Nickel Chef</h1>
+<p>
+This is the
+<a href="http://udacity.com/cs101x/urank/kathleen.html">
+best Hummus recipe!
+</a>
+
+</body>
+</html>
+
+
+
+
+
+
+""",
+   'http://udacity.com/cs101x/urank/kathleen.html': """<html>
+<body>
+<h1>
+Kathleen's Hummus Recipe
+</h1>
+<p>
+
+<ol>
+<li> Open a can of garbonzo beans.
+<li> Crush them in a blender.
+<li> Add 3 tablesppons of tahini sauce.
+<li> Squeeze in one lemon.
+<li> Add salt, pepper, and buttercream frosting to taste.
+</ol>
+
+</body>
+</html>
+
+""",
+   'http://udacity.com/cs101x/urank/arsenic.html': """<html>
+<body>
+<h1>
+The Arsenic Chef's World Famous Hummus Recipe
+</h1>
+<p>
+
+<ol>
+<li> Kidnap the <a href="http://udacity.com/cs101x/urank/nickel.html">Nickel Chef</a>.
+<li> Force her to make hummus for you.
+</ol>
+
+</body>
+</html>
+
+""",
+   'http://udacity.com/cs101x/urank/hummus.html': """<html>
+<body>
+<h1>
+Hummus Recipe
+</h1>
+<p>
+
+<ol>
+<li> Go to the store and buy a container of hummus.
+<li> Open it.
+</ol>
+
+</body>
+</html>
+
+
+
+
+""",
+}
+
+def get_page(url):
+    if url in cache:
+        return cache[url]
+    return ""
+
+
+def get_next_target(page):
+    start_link = page.find('<a href=')
+    if start_link == -1:
+        return None, 0
+    start_quote = page.find('"', start_link)
+    end_quote = page.find('"', start_quote + 1)
+    url = page[start_quote + 1:end_quote]
+    return url, end_quote
+
+def get_all_links(page):
+    links = []
+    while True:
+        url, endpos = get_next_target(page)
+        if url:
+            links.append(url)
+            page = page[endpos:]
+        else:
+            break
+    return links
+
+
+def union(a, b):
+    for e in b:
+        if e not in a:
+            a.append(e)
+
+def add_page_to_index(index, url, content):
+    words = content.split()
+    for word in words:
+        add_to_index(index, word, url)
+
+def add_to_index(index, keyword, url):
+    if keyword in index:
+        index[keyword].append(url)
+    else:
+        index[keyword] = [url]
+
+def lookup(index, keyword):
+    if keyword in index:
+        return index[keyword]
+    else:
+        return None
```

Comment:
This is what I meant in the previous comment. None means it doesn't exist.

Reply:
So I don't need to mention else?

Reply:
No, I mean this is the right thing to do, regarding the comment i made before.